source/common/grpc/async_client_manager_impl.h

```diff
@@ -49,19 +47,33 @@ class AsyncClientManagerImpl : public AsyncClientManager {
 public:
   AsyncClientManagerImpl(Upstream::ClusterManager& cm, ThreadLocal::Instance& tls,
                          TimeSource& time_source, Api::Api& api, const StatNames& stat_names);
+  RawAsyncClientSharedPtr
+  getOrCreateRawAsyncClient(const envoy::config::core::v3::GrpcService& config, Stats::Scope& scope,
+                            bool skip_cluster_check, bool always_use_cache = false) override;
```

Comment:
Again let's use an enum here, and I wouldn't give it a default. I don't think it's obvious what the default should be.

My suggestion would be to define the enum as
```
enum class CacheBehavior {
  AlwaysUseCache,
  CacheIfRuntimeEnabled
};
```